### PR TITLE
Remove boost.container_hash dependency

### PR DESCRIPTION
**Pull request by mborland**

### CODE REVIEW

**Comment by mborland** [comment]
Remaining dependencies can be [found here](https://github.com/mborland/multiprecision/issues/1). With math tracking towards being consumed in isolation it seems like it would be useful if multiprecision only depended on math.

File: include/boost/multiprecision/detail/hash.hpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+#include <cstddef>
+#include <functional>
+#include <system_error>
+
+namespace boost { namespace multiprecision { namespace detail {
+
+inline void hash_combine(std::size_t& seed) 
+{ 
+    (void)seed; // suppress warning: unused parameter 'seed'
```
Comment (jzmaddock) [inline include/boost/multiprecision/detail/hash.hpp:17]: Point of pedantry: couldn't we just not give the parameter a name?

File: include/boost/multiprecision/detail/hash.hpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+inline void hash_combine(std::size_t& seed, const T& v, Args... args) 
+{
+    std::hash<T> hasher;
+    seed ^= hasher(v) + 0x9e3779b9 + (seed<<6) + (seed>>2);
+    hash_combine(seed, args...);
+}
+
+inline std::size_t hash_value(std::error_condition const& v) 
+{
+    std::size_t seed = 0;
```
Comment (jzmaddock) [inline include/boost/multiprecision/detail/hash.hpp:30]: Point of pedantry #2: do we need this specialization?  It would remove one #include.

**Comment by mborland** [comment]
This should be good to go. The only build failure is `Received request to deprovision: The request was cancelled by the remote provider.`

**Comment by jzmaddock** [comment]
Thanks Matt!

**Comment by jzmaddock** [comment]
This is weird, no idea why it didn't show up in the PR CI tests: https://app.circleci.com/pipelines/github/boostorg/multiprecision/609/workflows/bda88c58-8a63-4ce5-9f70-aa88102fd551/jobs/1834

**Comment by mborland** [comment]
Should be resolved by https://github.com/boostorg/multiprecision/pull/309

**Pull request merged**